wire-ios-data-model/Tests/MLS/ProteusToMLSMigrationCoordinatorTests.swift

```diff
@@ -60,6 +66,52 @@ class ProteusToMLSMigrationCoordinatorTests: ZMBaseManagedObjectTest {
         super.tearDown()
     }
 
+    // MARK: - Tests
+
+    func testMigrateOrJoinGroupConversations_CallsJoinGroupForNewConversations() async throws {
+        // GIVEN
+        let selfUser = ZMUser.selfUser(in: syncMOC)
+        selfUser.teamIdentifier = .init()
+        let conversation = ZMConversation.insertNewObject(in: uiMOC)
+        conversation.mlsGroupID = .random()
+        let groupID = try XCTUnwrap(conversation.mlsGroupID)
+        let mockMLSService = MockMLSService()
+        mockMLSService.mockConversations = [conversation]
+        mockMLSService.conversationExistsResponses[groupID] = false
+
+        let expectation = XCTestExpectation(description: "joinGroup should be called")
+
+        // WHEN
+        sut.migrateOrJoinGroupConversations()
+
+        Task {
+            try? await mockMLSService.joinGroup(with: groupID)
+            expectation.fulfill()
+        }
```

Comment:
Your test needs to verify that `sut.migrateOrJoinGroupConversations()` is calling `joinGroup`.
But here what is happening is that you're explicitly calling `joinGroup`  and fulfilling your expectation **from your test**
Even if `migrateOrJoinGroupConversations` was doing nothing, your test would still succeed.

Reply:
Updated the test!